internal/datanode/buffer.go

```diff
@@ -143,8 +146,9 @@ func (bm *DelBufferManager) CompactSegBuf(compactedToSegID UniqueID, compactedFr
 	}
 
```

Comment:
The reason for previous panic is that when calling method bm.LoadAndDelete, the function rollDelete will be invoked, which will set DelBuff.DelData equal to null. This logic will result in segmentFault in the process of merging SegDelBuf lately

Reply:
So we simply change LoadAndDelete to Load-Then-Delete. The issue will be fixed